### PR TITLE
Reset password bug fix.

### PR DESCRIPTION
**Pull request by gerard1001**
**What was fixed in his PR?**

As **a driver and operator**, I want to be able to reset my password in case I forgot it so that I can log in to the app.

**Requirements**:

- The user should be able to provide their email.
- The driver or operator should be able to receive a reset password link in their email.
- The driver or operator should be able to reset the password( i.e. provide a new password).
- The driver or operator should be able to login with the new password.
- The page is responsive.
- Appropriate error messages are displayed on unsuccessful attempts

How to test this work:

1. clone the backend develop branch. Use this [link](https://github.com/atlp-rwanda/Dominators-Phantom-BE.git).
2. Next clone the frontend develop from [here](https://github.com/atlp-rwanda/Dominators-Phantom-FE.git) and the checkout to  the new branch "bug-fix-ft-Reset-Pwd".
3. Install all dependencies with with yarn install.
4. Run yarn start dev to start.
5. Test the reset password and login functionality.

**Extra things to consider:**

1. You might need to include in your frontend ._env_ file the following: 
      _BACKEND_URL=http://localhost:3000/api/v1_

2.  Also check for these keys and values in your backend .env file:
      _EMAIL_CRYPT=sudo
      FRONTEND_URL=http://localhost:5000
      APP_MAIL= dominatorsandela@gmail.com
      APP_MAIL_PASSWORD= smnfmcuksnwhckkx
      USER_EMAIL= dominatorsandela@gmail.com
      USER_EMAIL_P= smnfmcuksnwhckkx_

### CODE REVIEW

**Review by fab-ryan — COMMENTED** [review]
remove those console.log

**Review by UmuhireJessie — APPROVED** [review]

File: src/components/ResetPassword/ResetLinkComponent.js [hunk 1]
```diff
@@ -28,21 +28,17 @@ const ResetPasswordLink = () => {
...
+        if (result.Message == "success") {
           toast.success(result.record);
-          console.log("THIS IS A RESULT", result);
-        } else if (result.status == "fail") {
+        } else if (result.Message == "fail") {
           if (result.code == 400) {
             toast.error(result.record, { theme: "colored" });
-            console.log("THIS IS A RESULT", result);
           } else if (result.code == 401) {
             toast.error(result.record, { theme: "colored" });
```
Comment (Anguandia) [inline src/components/ResetPassword/ResetLinkComponent.js]: These 2 conditions are the same, please just remove the duplication, just toast the error message

File: src/index.js [hunk 2]
```diff
@@ -13,14 +13,14 @@ const root = reactDom.createRoot(document.getElementById("root"));
 
 root.render(
   <Provider store={store}>
-    <RoleProvider>
-      <RoleDetailsProvider>
-        <PermissionProvider>
+    {/* <RoleProvider> */}
+      {/* <RoleDetailsProvider> */}
+        {/* <PermissionProvider> */}
```
Comment (Anguandia) [inline src/index.js]: Did you consult with @tjeaneric and @UmuhireJessie before commenting out roles & permissions-related code? if so please remove the commented-out code then

**Review by Anguandia — CHANGES_REQUESTED** [review]
Great job @gerard1001 , please address the following:
- In the PasswordResetLink component, I think it makes sense to display a reset link sent message if successful, that display the same input field and button again
- The existing USER_EMAIL and USER_EMAIL_P are sufficient, we do not need APP_MAIL and APP_MAIL passwords in the backend, this is being fixed from the backend. Please review that in the PR description
- It would be good to auto-adjust the dimensions of the password-reset related forms according to their content. One styling file would do
- Please remove any commented-out code in files changed by your PR

File: src/components/ResetPassword/ResetLinkComponent.js (hunk 1 above)
Comment (gerard1001) [reply src/components/ResetPassword/ResetLinkComponent.js]: Thanks Mike. The issues are fixed now.

File: src/index.js (hunk 2 above)
Comment (gerard1001) [reply src/index.js]: I just returned them back the way they were.

**Review by Anguandia — APPROVED** [review]

**Review by furebo — APPROVED** [review]

**Pull request merged**